StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift

```diff
@@ -127,15 +127,30 @@ public final class EmbeddedPaymentElement {
                 return UpdateResult.canceled
             }
 
-            // 2. Re-initialize embedded view to update the UI to match the newly loaded data.
+            
+            // 2.1. Re-initialize embedded form view controller to update the UI to match the newly loaded data.
+            var formViewController: EmbeddedFormViewController?
+            if let formPaymentMethodType = self?.formViewController?.selectedPaymentOption?.paymentMethodType {
+                formViewController = EmbeddedFormViewController(configuration: configuration,
+                                                                intent: loadResult.intent,
+                                                                elementsSession: loadResult.elementsSession,
+                                                                savedPaymentMethods: loadResult.savedPaymentMethods,
+                                                                paymentMethodType: formPaymentMethodType,
+                                                                previousPaymentOption: self?.formViewController?.selectedPaymentOption,
+                                                                analyticsHelper: analyticsHelper,
+                                                                formCache: .init()) // Always use a fresh cache after an update
+                
+            }
+            
+            // 2.4 Re-initialize embedded view to update the UI to match the newly loaded data.
             let embeddedPaymentMethodsView = Self.makeView(
                 configuration: configuration,
                 loadResult: loadResult,
                 analyticsHelper: analyticsHelper,
-                previousPaymentOption: self?._paymentOption,
+                previousPaymentOption: formViewController != nil ? formViewController?.selectedPaymentOption : self?._paymentOption, // Use the possibly new payment option from the formVC if it exists, self?._paymentOption is not yet updated with the new formVC
```

Comment:
1. we should break this into a separate line
2. I'm not following what we're doing here, because the computed variable has:
```
        if let formPaymentOption = formViewController?.selectedPaymentOption {
            return formPaymentOption
        }
```
I'm sure i'm missing something, but it seems like we could just pass in self?._paymentOption without checking formViewController here.

Reply:
1. Broke this into it's own line
2. paymentOption looks at `self.formViewController`, but at this point in the function `self.formViewController` is not yet updated with the latest instance, `self.formViewController` is updated with this new instance later in the function with the rest of the properties. So we need to look at the freshly created formViewController if it exists for a previous payment option. I added a better comment here.